src/main/java/redis/clients/jedis/StreamConsumersInfo.java

```diff
@@ -0,0 +1,60 @@
+package redis.clients.jedis;
+
+import java.util.Map;
+
+/**
+ * This class holds information about a consumer
+ * They can be access via getters.
+ * For future purpose there is also {@link #getConsumerInfo()} ()}  method
+ * that returns a generic {@code Map} - in case where more info is returned from a server
+ *
+ */
+public class StreamConsumersInfo {
+
+  public final static String NAME = "name";
+  public final static String IDLE = "idle";
+  public final static String PENDING = "pending";
+
+
+  private final String name;
+  private final long idle;
+  private final long pending;
+  private final Map<String,Object> consumerInfo;
+
+  /**
+   * @param map contains key-value pairs with consumer info
+   *
+   */
+  public StreamConsumersInfo(Map<String,Object> map) {
+
+    if (map!= null && map.size()>0) {
+      consumerInfo = map;
+      name = (String) map.get(NAME);
+      idle = (long) map.get(IDLE);
+      pending = (long) map.get(PENDING);
+
+
+    } else throw new IllegalArgumentException();
```

Comment:
If we are throwing runtime exception anyway when map is null isn't it better to just let it throw NullPointerException?

Reply:
Done. My initial thought was that it is not nice to see NullPointers